Jenkins_jobs/UpdateMachineIdentifiers.groovy

```diff
@@ -0,0 +1,73 @@
+@Library('NodeHelper') _
+
+node {
+    String[] machineNames = params.machineNames.trim().split(",")
+    def nodeHelper = new NodeHelper();
+
+    /* if the flag updateDescription is not set we only update the labels
+     * The user can chose to either overwrite, add preconfigured labels or append labels
+     *
+     * To overwrite labels, the overwrite flag must be set
+     * To add preconfigured labels, only the manchine names need to passed in
+     * To append labels, a set of label(s) must be passed in
+     */
+    stage('Update_Labels') {
+        String[] labels;
+        if (!params.labels.isEmpty()) {
+            labels = params.labels.trim().split(",")
+        }
+
+        for (int index = 0; index < machineNames.length; index++) {
+            println "Machine ${machineNames[index]} old labels: ${nodeHelper.getLabels(machineNames[index])}"
+
+            if (!params.overWriteLabels && params.projectLabel.isEmpty()) {
+                // We shoulidn't be touching any machine if a project label isn't is passed
+                error("Neither project label was provied nor overWriteLabels flag was set")
+            }
+
+            if (params.overWriteLabels && labels == null) {
+                error("No labels supplied for overWriteLabels option")
```

Comment:
why is this case an error? How to I overWrite with just preconfigured labels?

Reply:
it's to prevent erasing old labels if nothing is supplied. In order to over write preconfigured labels, you'll have to do a over write with the non preconfigured labels and then run the job with no labels and overWriteLabels option not set.
Ah I see the use now, initially the motivation behind having it this way was to prevent non preconfigured labels from being overwritten. But I see your point, I'll update the code